File: apps/Lodgr.Api/Models/District.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Lodgr.Api.Models;

[Table("districts")]
public class District
{
    [Key]
    [StringLength(20)]
    [Column("code")]
    public string Code { get; set; } = string.Empty;

    [Required]
    [StringLength(20)]
    [Column("province_code")]
    public string ProvinceCode { get; set; } = string.Empty;

    [Required]
    [StringLength(100)]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(150)]
    [Column("full_name")]
    public string FullName { get; set; } = string.Empty;

    [StringLength(50)]
    [Column("administrative_unit")]
    public string? AdministrativeUnit { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
}

File: apps/api/src/main/java/io/github/hoanghonghuy/lodgr/entity/address/District.java
package io.github.hoanghonghuy.lodgr.entity.address;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "districts")
@Getter
@Setter
public class District {
    @Id
    @Column(name = "code", length = 10, nullable = false)
    private String code;

    @Column(name = "code_name", nullable = false, length = 50)
    private String codeName;

    @Column(name = "code_name_en", length = 50)
    private String codeNameEn;

    @Column(name = "name", length = 100)
    private String name;

    @Column(name = "name_en", length = 100)
    private String nameEn;

    @ManyToOne
    @JoinColumn(name = "province_code")
    private Province province;

    @OneToMany(mappedBy = "district", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Ward> wards = new HashSet<Ward>();
}
